src/config/field-overrides/dataSet.js

```diff
@@ -52,39 +54,149 @@ class RenderAsTabsSettings extends React.Component {
                 : undefined
 
         this.props.onChange({ target: { value: renderAsTabs } });
-        this.updateTabsDirection(tabsDirection)
+        const newDisplayOptions = {
+            ...this.state.displayOptions,
+            tabsDirection
+        }
+        this.updateDisplayOption(newDisplayOptions)
+    }
+
+    onAddCustomTextChanged = (event) =>  {
+        const addCustomText = event.target.value
+        const customText =
+            addCustomText
+                ? {header: undefined, subheader: undefined, alignemt: 'center'}
+                : undefined
+
+        const newDisplayOptions = {
+            ...this.state.displayOptions,
+            customText
+        }
+        this.updateDisplayOption(newDisplayOptions)
+    }
+
+    onCustomTextAlignmentChanged = (event) =>  {
```

Comment:
You could also combine these `onCustomTextAlignmentChanged`, `onCustomTextHeaderChanged`, `onCustomTextSubheaderChanged`, by doing something like

```
    onCustomTextChanged = (event, field) =>  {
        const customText =
            {...this.state.displayOptions[field],  [field]: event.target.value}

        const newDisplayOptions = {
            ...this.state.displayOptions,
            customText
        }
        this.updateDisplayOption(newDisplayOptions)
    }
```
but I'm not sure if the reused code makes up for the redice readability

Reply:
went for an hybrid between the 2. let me know what you think!